src/puppetlabs/puppetdb/command.clj

```diff
@@ -270,10 +270,17 @@
   (let [id (str id)
         received-time (str (tcoerce/to-long received-time))
         duration (str (in-millis (interval start-time (now))))
-        command-name (command-names command-kw)]
-    (if-let [{:keys [puppet-version]} opts]
+        command-name (command-names command-kw)
```

Comment:
The changes in this file feel like they're unrelated to a producer-timestamp bug?

Reply:
You're right these changes are somewhat unrelated to the bug, but they do add logging to the behavior that bug was suppressing so I went ahead and added it in the same pr. I'm hoping it will help add clarity when people start seeing messages get bashed.